CMakeLists.txt

```diff
@@ -4,6 +4,8 @@
 
 cmake_minimum_required(VERSION 3.2)
 
+include(GNUInstallDirs)
```

Comment:
`EnTT` fully supports also VS. Isn't this inherently wrong in that case?

Reply:
Actually here this only makes some variables avialable where only `CMAKE_INSTALL_INCLUDEDIR` is used and which evaluates to `include`. This should be correct or okay for most platforms. The other variables from `GNUInstallDirs` are not needed because the library is header only. We could also omit the include and replace the variable with a simple `include` if prefered.

Reply:
Don't worry, often I make comments just to better understand what's going on. If you confirm that it works also on VS, it's fine for me.

Reply:
I confirm it works under VS. Just tested it for playing safe.

Reply:
Ok, I just noticed that I used also the ${CMAKE_INSTALL_LIBDIR} variable from `GNUInstallDirs` for the exported and installed config file. This also works like a charm on Windows (VS) though I install it to one of the most common path from the unix conventions: `<prefix>/(lib/<arch>|lib|share)/cmake/<name>*/`. But there seems to be different conventions where to install the config file on different platforms: https://cmake.org/cmake/help/v3.2/command/find_package.html

> CMake constructs a set of possible installation prefixes for the package. Under each prefix several directories are searched for a configuration file. The tables below show the directories searched. Each entry is meant for installation trees following Windows (W), UNIX (U), or Apple (A) conventions.

> This is merely a convention, so all (W) and (U) directories are still searched on all platforms.

Reply:
Is there a way to somehow _respect_ the conventions for the different platforms? `EnTT` has several users on Win after all.

Reply:
Yes. I'll try to adapt this to conform to the conventions dependent on the platform id using generator expressions.

Reply:
Good. Ping me if you find a solution. I'll leave the PR open while waiting for a feedback.  
Thank you for your efforts and your help to improve `EnTT`. Really appreciated.

Reply:
Okay generator expressions are not available in install for the destination so I had to use a variable `ENTT_INSTALL_CONFIGDIR`. The Windows convention for the location of the CMake config files should now be respected.

Reply:
Thank you, too. I would say it's ready to merge.

Just a little thing that I did not respect from the "modern CMake guidelines" is that they say one should "not abuse usage requirements: As an example, don’t add -Wall to the PUBLIC or INTERFACE section of target_compile_options, since it is not required to build depending targets." As this was already the case before, I did not change it.

Reply:
It's fine for me to keep on board `-Wall`. It helps quite a lot during development actually. The other way around that I could accept is to force `-Wall` only when compiling the tests. Do you think it's possible to do this?

Reply:
Yes you can add it to any target you want. So for the tests, should I add it only to the `odr` target or also to all the single test targets? By the way what stands odr for, One Definition Rule?

* Is there a reason you didn't add `-pedantic -Wall` for MSVC?
* I think the optimization and the libc++ options are also belonging to usage requirements. Do you want to keep them on board? - is libc++ improving performance?
* On my system (Archlinux with gcc 8.1.0 and cmake 3.11.2) the `-g` option is automatically appended when building in Debug mode so I get it twice. Also in Release mode it gives me the `-O3` optimization flag by default and I get it twice.

Reply:
>Is there a reason you didn't add -pedantic -Wall for MSVC?

If a reasons existed, I forgot it actually.

>I think the optimization and the libc++ options are also belonging to usage requirements. Do you want to keep them on board? - is libc++ improving performance?

If I remember it right, it was due to the fact that on mac it didn't compile without including libc++. Because of this I put a line to _prefer_ libc++ in general. We can rearrange it differently, not a problem. PR are compiled on the fly both on Travis and appveyor, so we know immediately if changes ruin something.

>On my system [...]

Well, I don't think we can assume this on all systems and putting twice those options isn't a problem, so...

---

In general, it looks like you are very skilled with cmake, so I tend to consider your suggestions.  
If you find we can further improve the build system of `EnTT`, feel free to continue pushing on the PR. We aren't in a hurry to merge it. ;-)

Reply:
I think for MSVC, we miss much things, especially as CMake Interface EnTT should use those flags on MSVC -> 
# /Zi - Produces a program database (PDB) that contains type information and symbolic debugging information for use with the debugger.
# /FS - Allows multiple cl.exe processes to write to the same .pdb file
# /DEBUG - Enable debug during linking
# /Od - Disables optimization
# /Ox - Full optimization
# /permissive- (c++ conformance)

Reply:
@skypjack Alright I think it's best to add the flags to the test targets then.
@Milerius I think these options/flags are no real requirements for using the library and so they don't have to be set as usage requirements on the EnTT target. But we can add some flags individually for the tests if needed. Projects depending on EnTT can define their own flags.

Reply:
@m-waka Fine for me. Do you plan to do it on your branch and therefore with this PR?

Reply:
@skypjack yes I will implement it on my branch with this PR tomorrow.